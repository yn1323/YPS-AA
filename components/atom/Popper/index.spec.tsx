import React from 'react'
import Component from '.'
import { COMMON_MENU } from '@constant/layout/menus'
import { disableWarn, mountWithTheme } from '@test/TestWrapper'

describe(`${Component.name}`, () => {
  test(`${Component.name}-snapshot`, () => {
    const args = {
      show: false,
      setShow: (condition: boolean) => {
        console.log(condition)
      },
      anchorEl: document.getElementsByTagName('body')[0],
    }

    const wrapper = mountWithTheme(
      <Component {...args}>
        <div />
      </Component>
    )
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({
      ...args,
      show: true,
      placement: 'bottom-end',
      timeout: 200,
    })
    expect(wrapper).toMatchSnapshot()
  })

  test(`${Component.name}-event`, () => {
    const args = {
      show: true,
      setShow: (condition: boolean) => {
        console.log(condition)
      },
      items: COMMON_MENU[0].items,
      anchorEl: document.getElementsByTagName('body')[0],
    }
    const wrapper = mountWithTheme(
      <Component {...args}>
        <div>hello</div>
      </Component>
    )
    wrapper.find('span').map(elem => elem.simulate('click'))
  })

  disableWarn()
})
