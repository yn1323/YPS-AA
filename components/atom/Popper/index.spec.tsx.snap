// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popper-snapshot 1`] = `
<Popper
  anchorEl={<body />}
  setShow={[Function]}
  show={false}
>
  <Popper__Container
    anchorEl={<body />}
    open={false}
    placement="bottom-start"
    transition={true}
  >
    <ForwardRef(Popper)
      anchorEl={<body />}
      className="Popper__Container-sc-fo3olu-0 ibMCCu"
      open={false}
      placement="bottom-start"
      transition={true}
    />
  </Popper__Container>
</Popper>
`;

exports[`Popper-snapshot 2`] = `
<Popper
  anchorEl={
    <body>
      <div
        class="Popper__Container-sc-fo3olu-0 ibMCCu"
        role="tooltip"
        style="position: absolute; top: 0px; left: 0px;"
        x-placement="bottom-end"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
          style="opacity: 1; webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <ul
            class="MuiList-root MuiList-padding"
            role="menu"
            tabindex="-1"
          >
            <div
              tabindex="0"
            />
          </ul>
        </div>
      </div>
    </body>
  }
  placement="bottom-end"
  setShow={[Function]}
  show={true}
  timeout={200}
>
  <Popper__Container
    anchorEl={
      <body>
        <div
          class="Popper__Container-sc-fo3olu-0 ibMCCu"
          role="tooltip"
          style="position: absolute; top: 0px; left: 0px;"
          x-placement="bottom-end"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
            style="opacity: 1; webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          >
            <ul
              class="MuiList-root MuiList-padding"
              role="menu"
              tabindex="-1"
            >
              <div
                tabindex="0"
              />
            </ul>
          </div>
        </div>
      </body>
    }
    open={true}
    placement="bottom-end"
    transition={true}
  >
    <ForwardRef(Popper)
      anchorEl={
        <body>
          <div
            class="Popper__Container-sc-fo3olu-0 ibMCCu"
            role="tooltip"
            style="position: absolute; top: 0px; left: 0px;"
            x-placement="bottom-end"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
              style="opacity: 1; webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
            >
              <ul
                class="MuiList-root MuiList-padding"
                role="menu"
                tabindex="-1"
              >
                <div
                  tabindex="0"
                />
              </ul>
            </div>
          </div>
        </body>
      }
      className="Popper__Container-sc-fo3olu-0 ibMCCu"
      open={true}
      placement="bottom-end"
      transition={true}
    >
      <ForwardRef(Portal)
        disablePortal={false}
      />
    </ForwardRef(Popper)>
  </Popper__Container>
</Popper>
`;
